pkg/services/ngalert/schedule/alert_rule.go

```diff
@@ -233,28 +233,37 @@ func (a *alertRule) Run(key ngmodels.AlertRuleKey) error {
 					a.evalApplied(key, ctx.scheduledAt)
 				}()
 
-				for attempt := int64(1); attempt <= a.maxAttempts; attempt++ {
```

Comment:
So, moving the start of the loop means that rule changes (updates, pauses, moving to new folder, ...) would not affect remaining retry attempts. I imagine the functional impact of this is fairly low / non-existent but just want to confirm that the intent.

Reply:
I did _pause_ a minute to think about this, I think it's reasonable that the state of the rule at the start of the evaluation is used for each attempt. If they're only seconds apart, then I don't know the benefit of having the rule pause/unpause only  seconds earlier, instead of waiting for the next tick.

Reply:
I've reverted the moving of the pause code out of paranoia, and to get this PR merged quicker, we can do it as a cleanup in a subsequent commit.